src/XrdHttp/XrdHttpMon.cc

```diff
@@ -33,14 +33,30 @@ void XrdHttpMon::RecordCount(XrdHttpReq::ReqType op, StatusCodes sc) {
 
     auto& info = statsInfo[op][sc];
     info.count++;
-    // info.duration.fetch_add(latency.count(), std::memory_order_relaxed);
 }
 
-void XrdHttpMon::RecordError(XrdHttpReq::ReqType op, StatusCodes sc) {
+void XrdHttpMon::RecordSuccess(XrdHttpReq::ReqType op, StatusCodes sc, std::chrono::steady_clock::duration duration) {
```

Comment:
When there are temporary changes before merging, they should be folded into the previous commits. I suggest adding first any required changes in XrdHttp not related to monitoring, followed by one commit with the XrdHttpMon implementation, and a final commit making use of the newly introduced interfaces in XrdHttp. See, for example, how XrdClCurl and XrdClS3 plugins are being integrated.